connor/main.go

```diff
@@ -0,0 +1,146 @@
+	}
+
+	marketClient := sonm.NewMarketClient(nodeCC)
+	taskCli := sonm.NewTaskManagementClient(nodeCC)
+	dealCli := sonm.NewDealManagementClient(nodeCC)
+	tokenCli := sonm.NewTokenManagementClient(nodeCC)
+
+	balanceReply, err := tokenCli.Balance(ctx, &sonm.Empty{})
+	if err != nil {
+		fmt.Printf("Cannot load balanceReply %v\r\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf(" >>>>> Balance 			   :: Live: %v, Side %v SNM \r\n", balanceReply.GetLiveBalance().Unwrap().String(), balanceReply.GetSideBalance().ToPriceString())
+	fmt.Printf(" >>>>> Eth Node Address    :: %s\r\n", cfg.Market.Endpoint)
+	fmt.Printf(" >>>>> Public Key address  :: %v\r\n", crypto.PubkeyToAddress(key.PublicKey).Hex())
+
+	ethAddr := &sonm.EthAddress{Address: crypto.PubkeyToAddress(key.PublicKey).Bytes()}
+	identityLVl := GetIdentityLvl(cfg)
+
+	dataUpdate := time.NewTicker(3 * time.Second)
+	defer dataUpdate.Stop()
+	tradeUpdate := time.NewTicker(3 * time.Second)
+	defer tradeUpdate.Stop()
+	poolTrack := time.NewTicker(900 * time.Second) //900
+	defer poolTrack.Stop()
+	poolInit := time.NewTimer(900 * time.Second)
+	defer poolInit.Stop()
+
+	snm := w.NewSNMPriceWatcher("https://api.coinmarketcap.com/v1/ticker/sonm/")
+	token := w.NewTokenPriceWatcher("https://api.coinmarketcap.com/v1/ticker/", "https://www.cryptocompare.com/api/data/coinsnapshotfullbyid/?id=")
+	reportedPool := w.NewPoolWatcher("http://178.62.225.107:3000/v1/eth/reportedhashrates/", []string{cfg.PoolAddress.EthPoolAddr})
+	avgPool := w.NewPoolWatcher("http://178.62.225.107:3000/v1/eth/avghashrateworkers/", []string{cfg.PoolAddress.EthPoolAddr + "/1"})
+	if err = snm.Update(ctx); err != nil {
+		log.Printf("cannot update snm data: %v\n", err)
+	}
+	if err = token.Update(ctx); err != nil {
+		log.Printf("cannot update token data: %v\n", err)
+	}
+	if err = reportedPool.Update(ctx); err != nil {
+		log.Printf("cannot update reportedPool data: %v\n", err)
+	}
+	if err = avgPool.Update(ctx); err != nil {
+		log.Printf("cannot update avgPool data: %v\n", err)
+	}
+
+	modules.ChargeOrdersOnce(ctx, marketClient, token, snm, balanceReply, cfg, ethAddr, identityLVl)
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Println("context done")
+			return
+
+		case <-dataUpdate.C:
+			if err = snm.Update(ctx); err != nil {
+				log.Printf(" cannot update SNM data: %v\n", err)
+			}
+			if err = token.Update(ctx); err != nil {
+				log.Printf("cannot update TOKEN data: %v\n", err)
+			}
+			go modules.CollectTokensMiningProfit(token)
+
+		case <-tradeUpdate.C:
+			modules.TradeObserve(ctx, ethAddr, dealCli, reportedPool, token, marketClient, taskCli, cfg, identityLVl)
+		case <-poolInit.C:
+			modules.SavePoolDataToDb(ctx, reportedPool, cfg.PoolAddress.EthPoolAddr)
+		case <-poolTrack.C:
+			modules.PoolTrack(ctx, reportedPool, avgPool, cfg.PoolAddress.EthPoolAddr, dealCli, marketClient)
+		}
+	}
+}
+
+func GetIdentityLvl(cfg *config.Config) sonm.IdentityLevel {
+	switch cfg.OtherParameters.IdentityForBid {
+	case 1:
```

Comment:
why not use a suitable type in the config itself?

Reply:
Fixed.

Reply:
where?)